### PR TITLE
Add a test for MODCLUSTER-785

### PR DESCRIPTION
**Pull request by jfclere**
New and make script more flexible.

### CODE REVIEW

File: test/native/MODCLUSTER-785/testit.sh
Comment (jajik) [inline test/native/MODCLUSTER-785/testit.sh:9]: Don't we want to include these tests in the global testsuite too?

Then with the [latest](https://github.com/modcluster/mod_proxy_cluster/commit/1625f7846fa42bc520bd73738317e47fa1fad1db) changes this would not be needed. I not I would suggest to use the same approach (i.e., if not defined, we should go with the default and inform the user).

File: test/native/MODCLUSTER-785/testit.sh
Comment (jajik) [inline test/native/MODCLUSTER-785/testit.sh:30]: I don't like those names. `LOCREPO` is more `ORIGINREPO` or `REPOORIGIN` or something like that, and `REPO` is not repository at all. (Maybe `MPCCONF` or `REMOTECONF`?).

File: test/native/MODCLUSTER-785/testit.sh
Comment (jajik) [inline test/native/MODCLUSTER-785/testit.sh:36]: There is no reason supply `HTTPD` variable when it is the same is the default. Also, wouldn't it be better to use `-d` instead of `&`.?

File: test/native/MODCLUSTER-785/testit.sh
Comment (jajik) [inline test/native/MODCLUSTER-785/testit.sh:128]: It would be good to stop and remove the running containers too.

**Review by jajik — COMMENTED** [review]
I have a few notes.

I think we should set up the global test script to run these as well. I'll look into that once we merge this one.

File: test/native/MODCLUSTER-785/testit.sh
Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:30]: I am bad with names... I will fix that.

File: test/native/MODCLUSTER-785/testit.sh
Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:36]: I will fix that.. But it is probably everywhere...

File: test/native/MODCLUSTER-785/testit.sh
Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:128]: Do we want to do that if the test fails? I guess no ;-)

File: test/native/MODCLUSTER-785/testit.sh
Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:36]: I don't get the HTTPD variable part...

Comment (jajik) [reply test/native/MODCLUSTER-785/testit.sh:36]: If you look into https://github.com/modcluster/mod_proxy_cluster/blob/main/test/native/httpd/Dockerfile, there is `ENV HTTPD=https://dlcdn.apache.org/httpd/httpd-2.4.57.tar.gz`, so you don't have to run it with `-e HTTPD=https://dlcdn.apache.org/httpd/httpd-2.4.57.tar.gz`.

File: test/native/MODCLUSTER-785/testit.sh
Comment (jajik) [reply test/native/MODCLUSTER-785/testit.sh:128]: I think we can let it running if it fails so that it's easy to check the logs.

Comment (rhusar) [reply test/native/MODCLUSTER-785/testit.sh:128]: Not cleaning up after running tests is an antipattern. Ideally, let this be controled by an optional 'debug' option.

Comment (rhusar) [reply test/native/MODCLUSTER-785/testit.sh:128]: Makes sense (please file an issue to track that), thanks.

File: test/native/MODCLUSTER-785/testit.sh
Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:36]: OK we might need HTTP as an environment variable to test old version of httpd like 2.4.53 ;-)

Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:36]: dlcdn.apache.org only has only the latest version normally.

Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:36]: I am also thinking of an option to run with a locally install httpd, which is what I need to debug MODCLUSTER-785.

Comment (jajik) [reply test/native/MODCLUSTER-785/testit.sh:36]: If it is due to the maintainer mode, you could try the image we run in CI. https://github.com/modcluster/ci.modcluster.io/tree/main/httpd-dev

Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh:36]: Actually the -d option is a BAD idea... the waitforhttpd waits for the nohup file...

File: test/native/MODCLUSTER-785/testit.sh
Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh]: Oops fixed.

Comment (rhusar) [reply test/native/MODCLUSTER-785/testit.sh]: BTW @jfclere please squash the 'oops' commit into the other commits

Comment (jfclere) [reply test/native/MODCLUSTER-785/testit.sh]: Done

**Comment by rhusar** [comment]
I would suggest to get into the habit of rebasing the branches before pusing - there is a reminder for that on github "This branch is out-of-date with the base branch" - basically the tests would have been run with old revision.

**Comment by jfclere** [comment]
@rhusar but the new test has to be run by hands for the moment, so that is OK for "this" PR.

**Comment by jajik** [comment]
@jfclere I believe everything except the rebase was solved (maybe we could try Update with Rebase button here). It would be nice to merge it so that I can then incorporate it into #120. :)

**Review by jajik — APPROVED** [review]
Approve with regard to the previous comment.

**Review by rhusar — APPROVED** [review]

**Comment by rhusar** [comment]
Thanks @jfclere and @jajik (and I am assuming you will cleanup some of the stuff from here in https://github.com/modcluster/mod_proxy_cluster/pull/120 ;-) ).

**Pull request merged**
